tensorflow/contrib/boosted_trees/lib/utils/sparse_column_iterable.cc

```diff
@@ -101,6 +101,11 @@ class IndicesRowIterator
     return (row_idx_ == other.row_idx_);
   }
 
+  bool operator<(const IndicesRowIterator& other) const {
+    QCHECK_LT(iter_, other.iter_);
```

Comment:
This should be QCHECK_EQ, to make sense, as in other operators.
Given that, how do you make sure this PR doesn't break things?
Can you add some test cases?

Reply:
A test would be a good idea. Another option would be to simply throw an exception (NotImplementedError) to clarify that we don't actually expect this to be used.